server/service/apple_mdm.go

```diff
@@ -6792,11 +6793,50 @@ func (svc *Service) RenewABMToken(ctx context.Context, token io.Reader, tokenID
 
 type getOTAProfileRequest struct {
 	EnrollSecret string `query:"enroll_secret"`
+	IdpUUID      string // The UUID of the mdm_idp_account that was used if any, can be empty, will be taken from cookies
+}
+
+func (getOTAProfileRequest) DecodeRequest(ctx context.Context, r *http.Request) (interface{}, error) {
+	enrollSecret := r.URL.Query().Get("enroll_secret")
+	if enrollSecret == "" {
+		return nil, &fleet.BadRequestError{
+			Message: "enroll_secret is required",
+		}
+	}
+
+	boydIdpCookie, err := r.Cookie(sharedMdm.BYODIdpCookieName)
+
+	if err == http.ErrNoCookie {
+		// We do not fail here if no cookie is found, we validate later down the line if it's required
+		return &getOTAProfileRequest{
+			EnrollSecret: enrollSecret,
+			IdpUUID:      "",
+		}, nil
+	}
+
+	if err != nil {
+		return nil, &fleet.BadRequestError{
+			Message:     "something went wrong parsing the boyd idp cookie",
+			InternalErr: err,
+		}
+	}
```

Comment:
You can simplify those multiple checks as `r.Cookie(...)` will only return `ErrNoCookie` as error (it is documented as such, it's not just from checking the code), which is just an unfortunate API decision from Go as you can simply check that the cookie is nil or not.

Reply:
Good catch, will merge these two.